internal/jobtokens/jobtokens.go

```diff
@@ -47,14 +47,14 @@ func New(n int) (*JobTokens, error) {
 
 // Take is how a job (goroutine) can Take its turn.
 func (jt *JobTokens) Take() {
-	jt.wg.Add(1)
 	<-jt.tokens
+	jt.wg.Add(1)
 }
 
 // Give is how a job (goroutine) can give back its turn once done.
 func (jt *JobTokens) Give() {
-	jt.wg.Done()
 	jt.tokens <- struct{}{}
+	jt.wg.Done()
```

Comment:
Good catch. This change made me realize the relationship between the token channel and the waitgroup better. As described in my previous comment, a job must relinquish its token before signalling it is done to the internal waitgroup.